osquery/tables/system/linux/pci_devices.cpp

```diff
@@ -275,11 +391,10 @@ QueryData genPCIDevices(QueryContext& context) {
   struct udev_list_entry *device_entries, *entry;
   device_entries = udev_enumerate_get_list_entry(enumerate.get());
 
-  auto del_udev_device = [](udev_device* d) { udev_device_unref(d); };
   udev_list_entry_foreach(entry, device_entries) {
     const char* path = udev_list_entry_get_name(entry);
-    std::unique_ptr<udev_device, decltype(del_udev_device)> device(
-        udev_device_new_from_syspath(udev_handle.get(), path), del_udev_device);
+    std::unique_ptr<udev_device, std::function<void(udev_device*)>> device(
```

Comment:
You could use `decltype(delUdevDevice)` as a type of deleter. Which is a bit simpler and more clear.

Reply:
Whops.  For some reason I thought you could only use `decltype` on lamba's.  Will change :+1: